src/util/TypeList.impl.hxx

```diff
@@ -0,0 +1,53 @@
+
```

Comment:
The reason I named it .impl instead of .inl is because it's not implementing the inlined members, but creating a separate _impl namespace which is used for template metaprogramming. I don't mind changing it tho, so I'll do it

Reply:
done